공통 컴포넌트 구현

mj0107 opened the pull request:
## 개요
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
1. Checkbox, CheckboxGroup 컴포넌트 구현
2. FavoriteButton(찜하기 버튼) 컴포넌트 구현
3. Input 컴포넌트 구현
4. RestaurantFilter 컴포넌트 구현

_자세한 구현사항은 컴포넌트 내 작성된 JSDoc과 코드를 보고 리뷰해주시면 감사하겠습니다!_

- 전체적으로 넘어오는 API 응답 데이터 형식에 따라 컴포넌트 수정이 조금씩 있을 것 같습니다!
- 폰트는 아직 적용하지 않은 상태입니다.
- 색은 추후에 값이 정해지면 수정하겠습니다!
- 필터 리셋 기능은 추후에 데이터 형식이 정해지고 나서 상태에 관한 구조를 보고 추가할 예정입니다.

## PR 유형
어떤 변경 사항이 있나요?
- [x] 새로운 기능 추가
- [x] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [x] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.

### Input
**default**
![image](https://github.com/user-attachments/assets/2e14a1da-b1d7-4b4a-a885-adc89eda122b)

**값 입력시**
![image](https://github.com/user-attachments/assets/9a9c6dac-7b95-47d2-8c18-f24f3b58c2b1)

### RestaurantFilter
![image](https://github.com/user-attachments/assets/f31ddca8-891d-4c00-81ec-d13cc84f3ae7)

### FavoriteButton
**찜하기를 하지 않았을 때**
![image](https://github.com/user-attachments/assets/08e2b5f0-9a88-4d5b-b760-9fee6458691a)

**찜하기를 했을 때**
![image](https://github.com/user-attachments/assets/3fababd0-be48-4760-94e9-a86afb409954)

mj0107 committed:
fix: eslint 에러 해결

- prettier와 OS간 개행 방식 차이로 인한 `CR` 에러로 인해 자동으로 맞추도록 설정 추가로 해결
- Props에서 optional로 받을 시, 따로 지정을 해주지 않았다는 에러를 eslint에서 무시하도록 하여 해결

mj0107 committed:
design: 체크박스 이미지 추가

mj0107 committed:
feat: 공통 Input 컴포넌트 구현

mj0107 committed:
feat: 공통 Checkbox 컴포넌트 구현

mj0107 committed:
feat: 식당 필터 컴포넌트 구현

mj0107 committed:
feat: 찜하기 컴포넌트 구현

talentforest commented on src/assets/images/check.png:
react-icons에서 바로 아이콘 가져올 수 있지 않나요? 혹시 어떤 이유가 있으신지..??

mj0107 replied:
checkbox가 기존의 html에서 지원하는 checkbox랑 디자인이랑 달라서 기존의 checkbox 스타일을 아예 다 지워버리고 새로 작업이 필요했습니다.

기존의 스타일을 다 지우면, border, outline, 색깔, 테두리 등 아예 아무것도 없는 하얀 element가 되어서 check 했을 때 안했을때를 style을 일일이 다 지정해줘야 했습니다.
그래서 check 했을 때 배경을 파란색으로 바꾸고, checkbox의 배경에 image를 사용하여서 했습니다.

혹시 icon을 써서 하시는 방법이 있을까요? 검색해봤는데 저 방법 위주로 쓰더라구요

talentforest replied:
아 제가 다른 거랑 착각했네요..!🥹 `<label>` 태그를 이용해서 아이콘을 쓰는 방법이 있었던 것 같긴 한데 굳이 싶습니다. 넵 그대로 놔두셔도 될 것 같아용!

talentforest commented on src/components/common/FavoriteButton.tsx:
```diff
@@ -0,0 +1,34 @@
+import { BsHeart, BsHeartFill } from 'react-icons/bs';
+
+interface HeartButtonProps {
+  size: number | string;
```
`size` 속성은 디폴트 값을 정해놓고 이 부분은 옵셔널로 만드는 건 어떨까요?

mj0107 replied:
> `size` 속성은 디폴트 값을 정해놓고 이 부분은 옵셔널로 만드는 건 어떨까요?

좋습니다! 수정하겠습니다!

talentforest commented on src/components/common/FavoriteButton.tsx:
```diff
@@ -0,0 +1,34 @@
+ * @param onClick - 버튼 클릭 시 실행될 콜백 함수
+ */
+export default function FavoriteButton({
+  size,
```
여기에 `size = 26` 이런식으로 설정하면 26 사이즈인 아이콘은 `size` 속성을 전달 안해줘도 돼서 편할 것 같아요! 거의 아이콘은 사이즈가 26px이었던 것 같습니당.

mj0107 replied:
> 여기에 `size = 26` 이런식으로 설정하면 26 사이즈인 아이콘은 `size` 속성을 전달 안해줘도 돼서 편할 것 같아요! 거의 아이콘은 사이즈가 26px이었던 것 같습니당.

넵 요것도 수정하겠습니다!

talentforest commented on src/components/common/Input.tsx:
```diff
@@ -0,0 +1,80 @@
+  const textSize = textSizes[size];
+  const placeholderSize = `placeholder:${textSizes[size]}`;
+  const marginSize = marginSizes[size];
+  const paddingSize = paddingSizes[size];
```
이거는 그냥 className에서 조건부로 스타일 적용하는건 어떨까요? 그럼 굳이 객체로 관리할 필요가 없을 것 같아용.

```javascript
const smallStyle = size === "small" && "text-xs mr-2 px-[10px]"
const mediumStyle = size === "medium" && "mr-6 px-4"
```

이렇게 변수 선언해서 아래에 추가해도 될 것 같고 아니면 className 내부에서 삼항 연산자로 할수도 있을 것 같아요.

`${size === "small" ? "text-xs mr-2 px-[10px]" : "medium" && "mr-6 px-4"}`

talentforest replied:
아니면 그냥 size 속성은 빼는 건 어떨까요? `width` 값이 있기 때문에 `width` 속성에서 일정 값보다 크면 medium 스타일을 적용하는 식으로 할수도 있지 않을까요? 의견 어떠신지 궁금합니당

mj0107 replied:
> 이거는 그냥 className에서 조건부로 스타일 적용하는건 어떨까요? 그럼 굳이 객체로 관리할 필요가 없을 것 같아용.
> 
> ```js
> const smallStyle = size === "small" && "text-xs mr-2 px-[10px]"
> const mediumStyle = size === "medium" && "mr-6 px-4"
> ```
> 
> 이렇게 변수 선언해서 아래에 추가해도 될 것 같고 아니면 className 내부에서 삼항 연산자로 할수도 있을 것 같아요.
> 
> `${size === "small" ? "text-xs mr-2 px-[10px]" : "medium" && "mr-6 px-4"}`

사실 이건 지금은 small 밖에 없어서 그렇지만 추후에 혹시 large같은 크기가 필요할까봐 객체로 관리하고 있던건데 굳이 필요없을까요?
size 유무에 대한것도 많이 고민했었는데, size는 width, height로 물론 크기 조정도 되지만 정의된 스타일을 간편하게 지정하려고 넣어봤습니다.
말씀하신거처럼 size 빼고 해도 될 것 같아요!
근데, input 디자인 조정하고 연락주신다고 하셔서 이부분은 조금만 기다려보겠습니다!

talentforest commented on src/components/common/Input.tsx:
```diff
@@ -0,0 +1,80 @@
+        onBlur={() => setIsFocus(false)}
+        onKeyDown={e => handlePressEnter(e)}
+      />
+    </label>
```
오 `<label>` 태그로 `<input>`을 감싸기도 하는군요..! 하나 배워갑니당

mj0107 replied:
> 오 `<label>` 태그로 `<input>`을 감싸기도 하는군요..! 하나 배워갑니당

아이콘을 input에 들어간것처럼 보여야 했기에 아이콘을 클릭해도 input에 적용될 수 있도록 해봤습니다...!

talentforest commented on src/components/restaurant/RestaurantFilter.tsx:
```diff
@@ -0,0 +1,49 @@
+import Checkbox from '@/components/common/checkbox/Checkbox';
+import CheckboxGroup from '@/components/common/checkbox/CheckboxGroup';
+
+export default function RestaurantFilter() {
```
Filter라는 이름의 컴포넌트는 개별 필터만 있는 컴포넌트라고 헷갈릴수도 있을 것 같은데, 혹시 컴포넌트명을 RestaurantFilterForm은 어떠실까요...?

talentforest reviewed (commented):
우왕 너무 수고하셨어요!!! 여러가지 고민의 흔적이 느껴집니당...👍

mj0107 committed:
rename: 컴포넌트명 변경

- 기존 RestaurantFilter는 필터 하나만 나타내는 컴포넌트 명이라고 생각할 수 있어서 RestaurantFilterForm으로 변경

mj0107 committed:
refac: size prop을 optional로 변경

- 기존 size가 26이 많아, default value로 26으로 설정하고 size를 optional로 변경

mj0107 committed:
design: 색상 설정 반영

talentforest reviewed (commented):
일단 리뷰는 완료하겠습니다~
